Simplify signals

mightyCelu opened the pull request:
This PR simplifies the use of signals in Veins and makes it consistent:
- Signals are registered once, in the class they are provided by, i.e., usually emitted.
- The handles are public static constants
- Clients of the signal use this handle to interact with the signal, such that the constants/defines of signal names are unnecessary.

In addition, I removed `simsignalwrap_t`, which worked around a behavior that changed in OMNeT++ 4.3.

The patchset does not change the simulation fingerprint (` 01fc-29ee/tplx`) and passes the tests (`tests/traci`, `subprojects/catch`).

dbuse commented on src/veins/modules/mobility/traci/TraCIVehicleInserter.h:
```diff
@@ -23,10 +23,11 @@
 namespace Veins {
 
-class TraCIScenarioManager;
+//class TraCIScenarioManager;
```
Shouldn't this line be deleted instead?

dbuse commented on src/veins/modules/mobility/traci/TraCIScenarioManager.h:
```diff
@@ -63,11 +63,11 @@ class TraCICommandInterface;
+    static const simsignal_t traciModuleAddedSignal;
+    static const simsignal_t traciModuleRemovedSignal;
+    static const simsignal_t traciTimestepBeginSignal;
+    static const simsignal_t traciTimestepEndSignal;
```
Are `static const simsignal_t`s really the way to go? If so, cool!
Previously I just used them as default-initialized private members. That also keeps them out of constructor lists and would avoid any problems coming along with external linkage (the typical issues of const data members). Plus: For a pseudo-singleton like the `TraciScenarioManager`, it should not be a performance problem at all.
Though that way would require all using classes to keep a copy of the `simsignal_t` of their own... so maybe the `ststic const` way is the best. I just am not sure about all the implications.

mightyCelu commented:
- You're right about the phrasing, I'll change it, however it only affects the commit messages if I'm not mistaken.
- I do not think the string constants serve a purpose anymore. All use-cases I can think of are possible by using the signal's handle that is accessible from the 'owning' module. If there is a use-case, one can derive the signal's name from that handle. So if there is no common scenario where there is a need to use the name, I don't think there is an advantage of using constants for the signal names, as they are only used in a single place in the code. Actually, the names are used in two places, one of them being the .ned file of the 'owning' module, but in that case, one cannot use a constant, anyway.

dbuse commented:
Agreed. Then IMHO this is ready for merging given the tiny update to the commit messages. But maybe @sommer can to that when merging anyway.

mightyCelu commented:
I updated the PR to include the remarks. Additionally, I removed some duplicate signals that I missed in the first commits (Included in https://github.com/sommer/veins/pull/99/commits/1b87777a005801860bbfc4671a9d544ae4fed4ac) and renamed some signals to make the naming consistent.
